src/tools/dev/scripts/bv_support/bv_pidx.sh

```diff
@@ -221,6 +221,23 @@ function build_pidx
 #        else
 #            export LD_LIBRARY_PATH="$VISITDIR/mpich/$MPICH_VERSION/$VISITARCH/lib":$LD_LIBRARY_PATH
 #        fi
+    elif [[ "$parallel" == "yes" ]]; then
```

Comment:
I looked at Conduit, Adios2, MOAB and pidx and Conduit, Adios2 and pidx all use CMake, so those three could potentially be unified. The reason I started mucking around with the parallel libraries was because Conduit and pidx failed to build on Cori. The solution I came up with is pretty similar between the two. For Adios2, I'm not sure why that didn't have a problem. The only thing it is doing special for parallel is using the parallel compilers. It doesn't appear to be setting anything special otherwise, so at this point I'm even skeptical that it is really parallel. I would think it would need to look for some preprocessor variable that is being defined by the parallel compiler to do things differently in the parallel case. That is certainly possible, there are lots of really "clever" programmers out there.

I'm a little gun shy of making big changes here for a point release. I'm already worried about the fallout on other platforms and trying to fix this now could potentially lead to a multi-day rabbit hole, time which I don't have at the moment. Perhaps, we could file a ticket for this and move on.

Reply:
Just wanna confirm...were conduit, Adios2 and pidx plugins tested in parallel on Cori?

Reply:
I should have made this an or statement. I think if any one appears to work, its reasonable to believe the others will too. I will nevertheless still be perplexed as to how ;)

Reply:
We don't actually have any test data for Adios2 or pidx, so I have no way to test those.

Conduit has an mpi dependency from conduit relay. I looked at the xml file for the plugin and conduit relay is getting linked into the parallel engine. It vaguely remember Cyrus telling me it isn't being used, but is there in case we want to at some point. This all seems reasonable to me. Conduit (by way of the Blueprint plugin) is being tested in serial and parallel in the nightly test suite. I tested with one of the braid Blueprint files in serial and parallel on Cori and I got pictures, so I believe it works on Cori as well as quartz.

Reply:
@brugger1 @markcmiller86 When mpi is needed -- conduit provides additional libraries that provide mpi support and depend on mpi, but there is only one conduit install. 


We are using conduit's mpi lib, but only in in the  parallel engine:

```
    <LIBS components="EPar">
      conduit_relay_mpi
    </LIBS>
```

That said, adding:

```
    cfg_opts="${cfg_opts} -DCMAKE_C_FLAGS:STRING=\"${C_OPT_FLAGS} ${PAR_LINKER_FLAGS}\""
    cfg_opts="${cfg_opts} -DCMAKE_CXX_FLAGS:STRING=\"${CXX_OPT_FLAGS}\""	    cfg_opts="${cfg_opts} -DCMAKE_CXX_FLAGS:STRING=\"${CXX_OPT_FLAGS} ${PAR_LINKER_FLAGS}\""
```

Seems like it might make the serial libs also depend on mpi. 

I suspect we only needed these changes:

```
   if [[ "$PAR_INCLUDE" != "" ]] ; then
        cfg_opts="${cfg_opts} -DMPI_C_INCLUDE_PATH:STRING=${PAR_INCLUDE_PATH}"
        cfg_opts="${cfg_opts} -DMPI_CXX_INCLUDE_PATH:STRING=${PAR_INCLUDE_PATH}"
    fi

     if [[ "$PAR_LIBS" != "" ]] ; then
        cfg_opts="${cfg_opts} -DMPI_C_LINK_FLAGS:STRING=${PAR_LINKER_FLAGS}"
        cfg_opts="${cfg_opts} -DMPI_C_LIBRARIES:STRING=${PAR_LIBRARY_LINKER_FLAGS}"
        cfg_opts="${cfg_opts} -DMPI_CXX_LINK_FLAGS:STRING=${PAR_LINKER_FLAGS}"
        cfg_opts="${cfg_opts} -DMPI_CXX_LIBRARIES:STRING=${PAR_LIBRARY_LINKER_FLAGS}"
    fi
```

Reply:
> conduit provides additional libraries that provide mpi support and depend on mpi, but there is only one conduit install.

Awesome! That is a great way of doing business. I wish HDF5 did same. For HDF5, and I suspect other libs may behave similarly, we have either different paths to same named libs for serial and parallel or we have differently named libs at the same path.

In any event, the logic to link a VisIt `xxx_ser.so` target is different from a `xxx_par.so` target in terms of the paths and/or libs that are involved. And, I wasn't seeing anything suggestive of that distinguishment in this logic.

Reply:
I think that is solved with this guy in Blueprint xml:

```
   <LIBS components="EPar">
      conduit_relay_mpi
    </LIBS>
```

?

Reply:
> I think that is solved with this guy in Blueprint xml:
> 
> ```
>    <LIBS components="EPar">
>       conduit_relay_mpi
>     </LIBS>
> ```
> 
> ?

Ok, thats it! Finally, things a starting to make sense to me ;) I wonder if we have similar way of doing business for Adios2 and Pidx